frontend/packages/console-dynamic-plugin-sdk/src/app/modal-support/ModalProvider.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
+type CloseModal = () => void;
+
+export type ModalComponent = React.FC<{
+  closeModal: CloseModal;
+  [key: string]: unknown;
+}>;
+
+export type LaunchModal = (component: ModalComponent, extraProps: Record<string, unknown>) => void;
+
+type ModalContextValue = {
+  launchModal: LaunchModal;
+  closeModal: CloseModal;
+};
+
+const defaultModalContext: ModalContextValue = {
+  launchModal: null as any,
+  closeModal: null as any,
+};
+
+export const ModalContext = React.createContext(defaultModalContext);
```

Comment:
I'd just inline the default context value here

```ts
export const ModalContext = React.createContext({
  launchModal: _.noop,
  closeModal: _.noop,
});
```

https://reactjs.org/docs/context.html#reactcreatecontext

> The `defaultValue` argument is **only** used when a component does not have a matching Provider above it in the tree. This default value can be helpful for testing components in isolation without wrapping them.